strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/booters/StorageBooter.java

```diff
@@ -63,12 +60,18 @@ public void initialize()
         {
             try
             {
-                final Configuration configuration = configurationManager.getConfiguration();
+                Configuration configuration = configurationManager.getConfiguration();
 
                 initializeStorages(configuration.getStorages());
 
                 Collection<Repository> repositories = getRepositoriesHierarchy(configuration.getStorages());
 
+                logger.info("Reloading Repository Configurations....");
```

Comment:
@anki2189 have you any thoughts how to avoid this "Reloading" ?

Reply:
@sbespalov As loaded repository instances are immutable, reloading was only option.
One approach to avoid reloading is load mutable instances and make them immutable after performing various checks.

Reply:
this would be a bit error prone as well. I mean that what if someone else would like to reload something in case of configuration changes? For me this is seems that the only way to fix it safety is to trigger whole Spring application context reloading in case of configuration changes. 
@anki2189  so what do you think?

Reply:
Triggering whole Spring application context at server startup seems fine. But suppose someone marked Repository `Out of Service` from UI , it will not be good idea to trigger whole spring application Context reload in that case.

Reply:
yes, in this case we have it on application startup.

Reply:
Guys, did you reach an agreement on the context reload? Is there anything more to be done here or can I proceed with merging? :)

Reply:
@sbespalov As `GroupRepositorySetCollector#collect()` method traverses the repository tree and only mark Repository `Out of Service ` if case of any non-existent repository and `StorageBooter#initializeRepository()` does the initialization work after checking repository status (`In Service`). So , reloading context will be little trickier to code and test in both approaches we discussed (`Exception and flag`), reloading config seems more clean approach as of now.